### PR TITLE
Update GSgnnNodeModelInterface interface

### PR DESCRIPTION
**Pull request by classicsong**
*Issue #, if available:*

*Description of changes:*
Both predictions and embed_dict are either tensors or dictionaries of tensors

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: python/graphstorm/model/node_gnn.py [hunk 1]
```diff
@@ -200,13 +203,22 @@ def append_to_dict(from_dict, to_dict):
...
+            # pred can be a Tensor or a dict of Tensor
+            # emb can be a Tensor or a dict of Tensor
             if isinstance(pred, dict):
                 append_to_dict(pred, preds)
+            else:
+                assert len(seeds) == 1, \
+                    f"Expect prediction results of multiple node types {label.keys()}" \
+                    f"But only get results of one node type"
+                ntype = list(seeds.keys())[0]
+                append_to_dict({ntype: pred}, preds)
```
Comment (zheng-da) [inline python/graphstorm/model/node_gnn.py:216]: is this possible that `emb` is a dict while `pred` is a tensor?

Comment (classicsong) [reply python/graphstorm/model/node_gnn.py:216]: We just assume, the return value of model.predict can be a tensor or a dict and we turn them into dicts.

**Comment by zheng-da** [comment]
do we need tests to trigger the code?

**Comment by classicsong** [comment]
> do we need tests to trigger the code?

Added

**Review by zheng-da — APPROVED** [review]

**Pull request merged**